dotnet/src/Connectors/Connectors.Ollama/Core/ServiceBase.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Http;
+using Microsoft.SemanticKernel.Services;
+using OllamaSharp;
+
+namespace Microsoft.SemanticKernel.Connectors.Ollama.Core;
+
+/// <summary>
+/// Represents the core of a service.
+/// </summary>
+public abstract class ServiceBase
+{
+    /// <summary>
+    /// Attributes of the service.
+    /// </summary>
+    internal Dictionary<string, object?> AttributesInternal { get; } = new();
+    internal readonly OllamaApiClient _client;
+
+    internal ServiceBase(string model,
+        Uri endpoint,
+        HttpClient? httpClient = null,
+        ILoggerFactory? loggerFactory = null)
+    {
+        Verify.NotNullOrWhiteSpace(model);
+        this.AttributesInternal.Add(AIServiceExtensions.ModelIdKey, model);
+
+        if (httpClient is not null)
+        {
+            httpClient.BaseAddress ??= endpoint;
```

Comment:
I don't think mutating the HTTP client by updating the base address and headers collection is a good idea because we don't know whether the same instance of the client is used elsewhere.